src/github.com/couchbaselabs/sync_gateway/base/bucket.go

```diff
@@ -105,6 +130,123 @@ func (bucket couchbaseBucket) StartTapFeed(args walrus.TapArguments) (walrus.Tap
+	urls := []string{bucket.spec.Server}
+	poolName := bucket.spec.PoolName
+	if poolName == "" {
+		poolName = "default"
+	}
+	bucketName := bucket.spec.BucketName
+
+	vbucketIdsArr := []uint16(nil) // nil means get all the vbuckets.
+
+	receiver := NewDCPReceiver()
+
+	maxVbno, err := bucket.getMaxVbno()
+	if err != nil {
+		return nil, err
+	}
+
+	startSeqnos := make(map[uint16]uint64, maxVbno)
+	vbuuids := make(map[uint16]uint64, maxVbno)
+
+	// GetStatsVbSeqno retrieves high sequence number for each vbucket, to enable starting
+	// DCP stream from that position.  Also being used as a check on whether the server supports
+	// DCP.
+	statsUuids, highSeqnos, err := bucket.GetStatsVbSeqno(maxVbno)
+	if err != nil {
+		return nil, errors.New("Error retrieving stats-vbseqno - DCP not supported")
+	}
+
+	if args.Backfill == walrus.TapNoBackfill {
+		// For non-backfill, use vbucket uuids, high sequence numbers
+		LogTo("Feed+", "Seeding seqnos: %v", highSeqnos)
+		vbuuids = statsUuids
+		startSeqnos = highSeqnos
+	}
+	receiver.SeedSeqnos(vbuuids, startSeqnos)
+
+	// Auth handler for DCP connection needs to support bucket name defaults
+	username, password := "", ""
+	if bucket.spec.Auth != nil {
+		username, password, _ = bucket.spec.Auth.GetCredentials()
+	}
+	auth := &dcpAuth{username, password, bucketName}
+
+	LogTo("Feed+", "Connecting to new bucket datasource.  URLs:%s, pool:%s, name:%s, auth:%s", urls, poolName, bucketName, auth)
+	bds, err := cbdatasource.NewBucketDataSource(urls, poolName, bucketName, "", vbucketIdsArr, auth, receiver, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	events := make(chan walrus.TapEvent)
+	impl := couchbaseDCPFeedImpl{bds, events}
+
+	if err = bds.Start(); err != nil {
+		return nil, err
+	}
+
+	go func() {
+		for dcpEvent := range receiver.EventFeed {
+			events <- dcpEvent
+		}
+	}()
+
+	return &impl, nil
+}
+
+func (bucket couchbaseBucket) GetStatsVbSeqno(maxVbno uint16) (map[uint16]uint64, map[uint16]uint64, error) {
+
+	stats := bucket.Bucket.GetStats("vbucket-seqno")
+	if len(stats) == 0 {
+		// If vbucket-seqno map is empty, bucket doesn't support DCP
+		return nil, nil, errors.New("vbucket-seqno call returned empty map.")
+	}
+
+	uuids := make(map[uint16]uint64, maxVbno)
+	highSeqnos := make(map[uint16]uint64, maxVbno)
+
+	// GetStats response is in the form map[serverURI]map[]
+	for _, serverMap := range stats {
+		for i := uint16(0); i < maxVbno; i++ {
+			// stats come map with keys in format:
+			//   vb_nn:uuid
+			//   vb_nn:high_seqno
+			//   vb_nn:abs_high_seqno
+			//   vb_nn:purge_seqno
+			uuidKey := fmt.Sprintf("vb_%d:uuid", i)
+			highSeqnoKey := fmt.Sprintf("vb_%d:high_seqno", i)
+
+			highSeqno, err := strconv.ParseUint(serverMap[highSeqnoKey], 10, 64)
+			if err == nil && highSeqno > 0 {
+				highSeqnos[i] = highSeqno
+				uuid, err := strconv.ParseUint(serverMap[uuidKey], 10, 64)
+				if err == nil {
+					uuids[i] = uuid
+				}
+			}
+		}
+		break
```

Comment:
I'm confused why this break is here.  Are we only looking at stats[0]?  I think in that case, it could be changed to:

```
serverMap := stats[0]  // we only care about first server
for i := uint16(0); i < maxVbno; i++ {
  ...
}
```

which would have the nice side effect of reducing one level of indentation.